### PR TITLE
Create initial repo structure

### PR DESCRIPTION
**Pull request by hakbailey**
#### What does this PR do?
Sets up the basic structure, dependency management, test framework, settings, and Make commands according to our Python practices and the specific use cases for this codebase.

#### How can a reviewer manually see the effects of these changes?
Follow the instructions in the README to run and verify this locally. Feedback on the instructions is welcome!

#### Includes new or updated dependencies?
YES

#### What are the relevant tickets?
https://mitlibraries.atlassian.net/browse/ENSY-65 -

#### Developer
- [x] All new ENV is documented in README
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer
- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

**Comment by hakbailey** [comment]
@zotoMIT let me know if the ECR repo or function names need to change in the Makefile.

**Review by zotoMIT — APPROVED** [review]
This looks correct.

**Review by ehanson8 — APPROVED** [review]
One non-blocking question, this was a very helpful template

File: tests/test_ping.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from lambdas.ping import lambda_handler
+
+
+def test_ping():
+    assert lambda_handler({}, {}) == "pong"
+
+
+def test_ping_always_pongs():
```
[the remark this answers is not in the document]
Comment (hakbailey) [reply tests/test_ping.py:8]: Well technically they are testing different things: that the lambda responds correctly when data is supplied, and that it responds correctly when no data is supplied. The first test does cover all the code, but this is a good example of tests covering all the edge cases and not just what's expected (Jeremy wrote these in the timdex-pipeline-lambdas repo, I just copied them :) ). It's good practice, and we should do it more. The places where we run into production errors are very often where we didn't test enough edge cases.

[the remark this answers is not in the document]
Comment (ehanson8) [reply tests/test_ping.py:8]: Thanks, that makes sense, I'll add a similar test when I'm doing corrections on my PR
